add quick access to controller learning wizard via options menu

mxmilkiib opened the pull request:
implements feature request from issue #12262 to provide quicker access to the midi controller learning wizard.

## changes

### options menu submenu
- added "controller learning wizard" submenu under options menu
- dynamically populated with enabled controllers only
- shows "no controllers enabled" when no controllers are enabled
- updates automatically when controllers are enabled/disabled

### navigation
- clicking a controller menu item opens preferences dialog
- automatically navigates to that specific controller's page
- expands the controllers tree and selects the correct item
- user can then click the "learning wizard" button on that page

### implementation
- wmainmenubar: added menu creation and signal emission
- mixxxmainwindow: connected signals, filters to show only enabled controllers
- dlgprefcontrollers: added method to navigate to specific controller
- dlgpreferences: added method to delegate to controllers dialog

follows the same architectural pattern as the existing vinyl control menu implementation for consistency.

fixes #12262

Commit:
add quick access to controller learning wizard via options menu

implements feature request from issue #12262 to provide quicker access to the midi controller learning wizard.

- added "controller learning wizard" submenu under options menu
- dynamically populated with enabled controllers only
- hides when no controllers are enabled (fixes Qt 6.2.3 issue)
- updates when controllers are enabled/disabled via mappingApplied signal

- clicking a controller menu item opens preferences dialog
- automatically navigates to that specific controller's page
- expands the controllers tree and selects the correct item
- user can then click the "learning wizard" button on that page

- wmainmenubar: added menu creation and signal emission
- mixxxmainwindow: connected signals, filters to show only enabled controllers
- dlgprefcontrollers: added method to navigate to specific controller
- dlgpreferences: added method to delegate to controllers dialog

- use ControllerManager::mappingApplied instead of devicesChanged for menu updates
- hide menu when empty instead of disabled action (fixes Qt 6.2.3 clickable issue)
- flag-based solution instead of disconnect/reconnect for cleaner code flow
- suppress preferences dialog show when wizard launched from menu

follows the same architectural pattern as the existing vinyl control menu implementation for consistency.

fixes #12262

ronso0 commented on src/controllers/dlgprefcontrollers.cpp:
```diff
@@ -286,3 +286,40 @@ void DlgPrefControllers::slotMidiThroughChanged(bool checked) {
+                                m_pDlgPreferences,
+                                &DlgPreferences::show);
+                    },
+                    Qt::SingleShotConnection);
```
why do we need to disconnect this?
when a new mappng has been added, the mapping needs to be saved/applied, no?

Commit:
fix controller wizard menu issues per ronso0 review

- replace mappingApplied signal with direct Controller::openChanged connections
  - fixes menu not updating after startup
  - connect/disconnect as controllers are added/removed
  - properly tracks per-controller open state
- keep setVisible workaround for Qt 6.2.3 (already correct)
- replace suppress flag with preferences dialog visibility check
  - more elegant solution per ronso0 feedback
  - checks actual dialog visibility state at wizard start
  - only emits mappingStarted/mappingEnded if dialog was visible

Commit:
remove std::as_const wrappers from range-based for loops

Commit:
fix controller wizard menu not updating after startup

Connect to ControllerManager::mappingApplied in addition to
devicesChanged. devicesChanged is only emitted once on startup
(per the comment in controllermanager.cpp); mappingApplied fires
after a controller is opened when a mapping is applied post-startup,
so the menu now updates correctly when controllers are enabled.

Also simplify openLearningWizard loop to range-for per review.

Commit:
Fix MIDI learning wizard not appearing when opened from prefs button

DlgControllerLearning is parented to DlgPrefController, which is a child
of DlgPreferences. Previously, mappingStarted() was emitted after show(),
causing DlgPreferences::hide() to cascade-hide the wizard immediately.

Emit mappingStarted() before creating and showing the wizard so the prefs
dialog is already hidden when the wizard appears.

Commit:
Fix clazy range-loop-detach warnings in dlgprefcontrollers.cpp
